data/transactions/logic/assembler.go

```diff
@@ -1565,52 +1563,28 @@ func (ops *OpStream) assemble(text string) error {
 	for scanner.Scan() {
 		ops.sourceLine++
 		line := scanner.Text()
-		line = strings.TrimSpace(line)
-		if len(line) == 0 {
-			ops.trace("%3d: 0 line\n", ops.sourceLine)
-			continue
-		}
-		if strings.HasPrefix(line, "//") {
-			ops.trace("%3d: // line\n", ops.sourceLine)
-			continue
-		}
-		if strings.HasPrefix(line, "#pragma") {
-			ops.trace("%3d: #pragma line\n", ops.sourceLine)
-			ops.pragma(line)
-			continue
-		}
 		fields := fieldsFromLine(line)
-		if len(fields) == 0 {
-			ops.trace("%3d: no fields\n", ops.sourceLine)
-			continue
-		}
-		// we're about to begin processing opcodes, so settle the Version
-		if ops.Version == assemblerNoVersion {
-			ops.Version = AssemblerDefaultVersion
-		}
-		if ops.versionedPseudoOps == nil {
-			ops.versionedPseudoOps = prepareVersionedPseudoTable(ops.Version)
-		}
-		var current []string
-		var next []string
-		next = fields
-		for current, next = processFields(next); len(current) > 0 || len(next) > 0; current, next = processFields(next) {
+		for current, next := processFields(fields); len(current) > 0 || len(next) > 0; current, next = processFields(next) {
 			if len(current) == 0 {
 				continue
 			}
 			opstring := current[0]
-			if len(opstring) == 0 {
-				continue
-			}
 			if strings.HasPrefix(opstring, "//") {
-				// semicolon inside comment is not counted as newline
+				ops.trace("%3d: comment\n", ops.sourceLine)
 				break
 			}
-			currentLine := strings.Join(current, " ")
-			if strings.HasPrefix(currentLine, "#pragma") {
+			if opstring == "#pragma" {
```

Comment:
This may break some of the tests, but I think a pretty reasonable rule would be that # directives have to go on their own line. Any strong thoughts on that?

Reply:
The rule now is "If you have a #pragma, it consumes the rest of the line, no matter what" so as we make that true of all "# directives" that should handle most of the ideas I've though of, where the expansion contains semi-colons.

I could see only allowing it at the start of lines though.